include/otf2xx/chrono/convert.hpp

```diff
@@ -113,9 +118,15 @@ namespace chrono
          */
         otf2::chrono::ticks operator()(time_point t) const
         {
-            assert(ticks_per_second == clock::period::den);
+            if (ticks_per_second == clock::period::den)
+            {
+                return ticks(
+                    otf2::chrono::duration_cast<otf2::chrono::duration>(t.time_since_epoch()).count());
+            }
+
+            const double factor = static_cast<double>(clock::period::den) / ticks_per_second;
```

Comment:
Maybe store this in the class instead of recalculating it on every call as `ticks_per_second` is never used. (and make the ctors explicit while doing that...)

Downside: The (seemingly) [common](https://github.com/tud-zih-energy/otf2xx/blob/6124b31874a3e06ee5a076662787a97ca852aab0/include/otf2xx/writer/global.hpp#L631) [case](https://github.com/tud-zih-energy/otf2xx/blob/20c61661cd85e45f3f23aeca4b4d17ac4e986ad6/include/otf2xx/writer/archive.hpp#L512) of `factor == 1` won't be "optimized". But that introduced a branch which might be worse...

Reply:
Yeah, done in https://github.com/tud-zih-energy/otf2xx/blob/experimental-convert-refactor/include/otf2xx/chrono/convert.hpp#L124